### PR TITLE
various fixes

### PR DESCRIPTION
**Pull request by jblom**
should fix:

```
2024-07-05 08:58:25,604|INFO|7|io_util|delete_input_file|191|Verifying deletion of input file: /data/input-files/2101608170158176431__NOS_JOURNAAL_-WON01513227.mp4
2024-07-05 08:58:25,620|INFO|7|io_util|delete_input_file|199|Deleted VisXP input file: /data/input-files/2101608170158176431__NOS_JOURNAAL_-WON01513227.mp4
2024-07-05 08:58:25,621|ERROR|7|io_util|delete_input_file|212|OSError while removing empty input file dirs
Traceback (most recent call last):
File "/src/io_util.py", line 207, in delete_input_file
os.removedirs(
File "<frozen os>", line 243, in removedirs
OSError: [Errno 22] Invalid argument: '.'
2024-07-05 08:58:25,622|INFO|7|worker|callback|79|applying IO on output went well, now finally saving to DANE index
2024-07-05 08:58:25,622|INFO|7|worker|save_to_dane_index|97|saving results to DANE, task id=21caef8d2ffae61c100f0090da592d526674c0f9
```

### CODE REVIEW

**Comment by jblom** [comment]
not sure yet why this timeout occurred (maybe it's not every time):

```
Traceback (most recent call last):
File "/src/.venv/lib/python3.11/site-packages/elasticsearch7/connection/http_urllib3.py", line 255, in perform_request
response = self.pool.urlopen(
^^^^^^^^^^^^^^^^^^
File "/src/.venv/lib/python3.11/site-packages/urllib3/connectionpool.py", line 799, in urlopen
retries = retries.increment(
^^^^^^^^^^^^^^^^^^
File "/src/.venv/lib/python3.11/site-packages/urllib3/util/retry.py", line 525, in increment
raise six.reraise(type(error), error, _stacktrace)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/src/.venv/lib/python3.11/site-packages/urllib3/packages/six.py", line 770, in reraise
raise value
File "/src/.venv/lib/python3.11/site-packages/urllib3/connectionpool.py", line 715, in urlopen
httplib_response = self._make_request(
^^^^^^^^^^^^^^^^^^^
File "/src/.venv/lib/python3.11/site-packages/urllib3/connectionpool.py", line 469, in _make_request
self._raise_timeout(err=e, url=url, timeout_value=read_timeout)
File "/src/.venv/lib/python3.11/site-packages/urllib3/connectionpool.py", line 358, in _raise_timeout
raise ReadTimeoutError(
urllib3.exceptions.ReadTimeoutError: HTTPConnectionPool(host='esc-es-http.dane-es.svc.cluster.local', port=9200): Read timed out. (read timeout=30)
2024-07-05 08:59:18,958|INFO|7|base|log_request_success|265|POST http://esc-es-http.dane-es.svc.cluster.local:9200/dane-index-visxp/_doc?refresh=true&routing=21caef8d2ffae61c100f0090da592d526674c0f9 [status:201 request:23.269s]
2024-07-05 08:59:18,958|INFO|7|base_classes|_ack_with_status_msg|297|Send ACK + msg back to queue (async)
```


@Veldhoen I did add a try, except around the `self.save_to_dane_index` in the `worker.py` to at least catch any errors related to saving back to dane

**Comment by jblom** [comment]
@Veldhoen I also updated to python 3.11 I hope you have it on your system?

File: .github/workflows/_test.yml
Comment (Veldhoen) [inline .github/workflows/_test.yml:1]: > @Veldhoen I also updated to python 3.11 I hope you have it on your system?
Great, no problem.

File: worker.py [hunk 1]
```diff
@@ -79,12 +79,15 @@ def callback(self, task: Task, doc: Document) -> CallbackResponse:
...
-            )
+            try:
+                self.save_to_dane_index(
+                    doc,
+                    task,
+                    get_s3_output_file_uri(get_source_id(input_file_path)),
+                    provenance=full_provenance_chain,
+                )
+            except Exception:
+                logger.exception("Failed to save to DANE index")
```
Comment (Veldhoen) [inline worker.py:90]: When catching a general Exception, at least report back the nature of the exception?

File: io_util.py
Comment (Veldhoen) [inline io_util.py:1]: The deletion of input was only about removing empty directories, right? Seems like it shouldn't cause issues. Except if a worker would stay 'up' for a long time processing very many source files.. So might be nice to create an issue for the record, to do this properly and also to put this function under test.

File: worker.py
Comment (Veldhoen) [inline worker.py:1]: The try/except is a good addition, I guess. I can't estimate the impact of not saving back to the DANE index in practice though. Should we still debug this further? In that case, make a separate issue?

**Review by Veldhoen — APPROVED** [review]
Some small suggestions, mostly to create separate issues if we might still need to address problems

File: worker.py (hunk 1 above)
Comment (jblom) [reply worker.py:90]: this was definitely a quick "fix". It takes some scrutiny & tests to figure out which exceptions can really occur.

File: worker.py
Comment (jblom) [reply worker.py:1]: the next worker reads the S3 output of this from the DANE Result (I guess it's still like that). If somehow the 2nd worker now derives the S3 URI (for its input) from the DANE document (the url field), saving a result back to DANE might not be necessary.

I assume saving the DANE result is still needed, so if it works, things should probably be fine.

File: io_util.py
Comment (jblom) [reply io_util.py:1]: I only removed some code that cleaned a series of empty dirs, which used to be created by the `dane-download-worker`

In the new situation where the workers download from S3 themselves, this cleanup code was not necessary and actually threw some errors

File: worker.py (hunk 1 above)
Comment (Veldhoen) [reply worker.py]: I've added the Exception's content to the log message, so that it will be slightly easier to understand what goes wrong if this happens

**Pull request merged**